src/ostree.jsx

```diff
@@ -346,50 +294,115 @@ const Deployments = ({ versions }) => {
+                  columns={columns}
                   variant="compact"
+                  gridBreakPoint="grid-lg"
                   rows={items} />
     );
 };
 
-const DeploymentDetails = (akey, info, packages, doRollback, doUpgrade, doRebase, inProgress, setError, error, setOpenedKebab, openedKebab) => {
-    let name = null;
-    if (info && info.osname) {
-        name = info.osname.v;
-        if (info.version)
-            name += " " + info.version.v;
-    }
+const isUpdate = (info) => {
+    return client.item_matches(info, 'CachedUpdate') && !client.item_matches(info, 'DefaultDeployment');
+};
 
-    const isUpdate = () => {
-        return client.item_matches(info, 'CachedUpdate') && !client.item_matches(info, 'DefaultDeployment');
-    };
+const isRollback = (info) => {
+    return !client.item_matches(info, 'CachedUpdate') && client.item_matches(info, 'RollbackDeployment');
+};
 
-    const isRollback = () => {
-        return !client.item_matches(info, 'CachedUpdate') && client.item_matches(info, 'RollbackDeployment');
-    };
+const isRebase = (info) => {
+    return !info.id && !client.item_matches(info, 'BootedDeployment', 'origin') && !client.item_matches(info, 'RollbackDeployment') &&
+        !client.item_matches(info, "DefaultDeployment");
+};
 
-    const isRebase = () => {
-        return !info.id && !client.item_matches(info, 'BootedDeployment', 'origin') && !client.item_matches(info, 'RollbackDeployment') &&
-            !client.item_matches(info, "DefaultDeployment");
-    };
+const ConfirmDeploymentChange = ({ actionName, bodyText, onConfirmAction }) => {
+    const Dialogs = useDialogs();
+
+    const actions = [
+        <Button key="confirm-action" variant="warning"
+            onClick={() => { onConfirmAction(); Dialogs.close() }}
+        >
+            {cockpit.format(_("$0 and reboot"), actionName)}
+        </Button>,
+        <Button key="cancel-action" variant="link" onClick={Dialogs.close}>
+            {_("Cancel")}
+        </Button>
+    ];
+
+    const titleContent = (
+        <Flex justifyContent={{ default: "justifyContentFlexStart" }} spacer={{ default: 'spaceItemsSm' }}
+            flexWrap={{ default: 'nowrap' }}
+        >
+            <FlexItem>
+                <ExclamationTriangleIcon className="pf-v5-u-warning-color-100" />
+            </FlexItem>
+            <FlexItem>
+                <Text>{`${actionName}?`}</Text>
+            </FlexItem>
+        </Flex>
+    );
 
-    let state;
+    return (
+        <Modal isOpen
+            id="confirm-modal"
+            title={titleContent}
+            position="top"
+            variant="small"
+            onClose={Dialogs.close}
+            actions={actions}
+        >
+            <Text>{bodyText}</Text>
+        </Modal>
+    );
+};
+
+const DeploymentDetails = (akey, info, packages, doRollback, doUpgrade, doRebase, inProgress, setError, error, Dialogs) => {
+    const version = info.version ? info.version.v : null;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231128-144720-91f0c34a-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)